synapse/handlers/federation.py

```diff
@@ -2178,14 +2233,31 @@ async def _persist_auth_tree(
             room_id, [(event, new_event_context)]
         )
 
-    async def _prep_event(
+    async def _calculate_event_context(
         self,
-        origin: str,
         event: EventBase,
-        state: Optional[Iterable[EventBase]],
-        auth_events: Optional[MutableStateMap[EventBase]],
-        backfilled: bool,
-    ) -> EventContext:
+        state: Optional[Iterable[EventBase]] = None,
+        auth_events: Optional[MutableStateMap[EventBase]] = None,
+    ) -> Tuple[EventContext, MutableStateMap[EventBase]]:
+        """
+        Calculate the context and auth events for a given event.
+
+        Args:
+            event: The event itself.
+            state: The state events to calculate the event context from.
+            auth_events:
+                Map from (event_type, state_key) to event
+
+                Normally, our calculated auth_events based on the state of the room
+                at the event's position in the DAG, though occasionally (eg if the
+                event is an outlier), may be the auth events claimed by the remote
+                server.
+
+                Also NB that this function adds entries to it.
+
+        Returns:
+             The event context and auth events.
+        """
         context = await self.state_handler.compute_event_context(event, old_state=state)
 
         if not auth_events:
```

Comment:
I don't think this stuff (up to [line 2278](https://github.com/matrix-org/synapse/pull/9800/files#diff-0bc92da3d703202f5b9be2d3f845e375f5b1a6bc6ba61705a8af9be1121f5e42R2278)) belongs here. It is magic that prepares `auth_events` for the calls to `_update_auth_events_and_context_for_auth` and `event_auth.check` and as such, belongs next to those calls in `_check_event_auth`.

(this leaves you with `_calculate_event_context` just being the call to `state_handler.compute_event_context`, so you can basically just replace it with inline calls to `state_handler.compute_event_context`)

Reply:
Thanks! This cleans it up quite a bit. I've made this change and updated the description to be clearer what this is doing.